### ISSUE TITLE
Cant use name starting with use in the React Hooks' function Name.

### ISSUE THREAD

**Issue by durgeshsatyam** [facebook/react#17695]
**Do you want to request a *feature* or report a *bug*?**
Report a bug.
**What is the current behavior?**
If we name any hook's change state function starting with use* (for eg... useMyState) it shows errors when calling my hook from any function other than the current main function and throws errors, but as we change this function's name to any other name (for eg... changeMyState) it starts working fine.
I think the error is in the compiler which checks for the errors in React.
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have dependencies other than React. Paste the link to your JSFiddle (https://jsfiddle.net/Luktwrdm/) or CodeSandbox (https://codesandbox.io/s/new) example below:**

```
const [myState, useMyState] = React.useState(false); // causes problem
const [myNewState, changeMyNewState] = React.useState(false); // works great
function setState (value) {
  useMystate(value); // throws error.
  changeMyNewState(value); // it works great as aspected.
}
```

**What is the expected behavior?**
The expected behavior should be no error in each of the above-mentioned cases.
**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**
I am using 16.12.0
It does not compile so platform-independent issue.
haven't tried in previous versions.

**Comment by vkurchatkin** [comment]
You shouldn’t be doing that. Names starting with `use` are reserved for hooks.

**Comment by durgeshsatyam** [comment]
I believe, there should be proper documentation saying name starting with  `use` are reserved for hooks and we should not use it.

**Comment by bvaughn** [comment]
This is not a "compiler" error, but a lint error. 🙂 Your code should actually run fine regardless of what you name your state updates function.

The lint rule that enforces hooks are used correctly identifies a hook as a function that has a name beginning with "use". It's unfortunate that we don't have an easier/better way of identifying hooks, but we don't. For what it's worth, we searched a lot of projects (on GitHub and within Facebook) to make sure this naming convention wouldn't conflict with pre-existing code.

So basically, this is working as designed. It's an unfortunate but known limitation. You can disable the lint rule (although I would advise against it). Best to just abide by the naming convention. This will probably also make your code easier to understand for people who are not familiar with it.

Cheers!
